src/lib.rs

```diff
@@ -338,6 +342,59 @@ impl ShaderBuilder {
 		}
 	}
 
+	/// parses 
+	/// 
+	/// `//!ifdef SOME_DEFINE`
+	/// `//!ifndef SOME_DEFINE` 
+	/// `//!else`
+	/// `//!endif`
+	/// 
+	/// similar to c or glsl
+	/// 
+	/// # Arguments
+	/// - `defines` - list of strings which equal the defines in wgsl
+	pub fn parse_defines(&mut self, defines: &[String]) -> &mut Self {
```

Comment:
I might have made a mistake by making this PR about C-style defines.
The bevy engine has a built-in preprocessor that takes a list of strings and removes parts of the code that are in invalid ifdef statements. Using one source file to create different shaders at run-time is the point.
Maybe it would make sense to rename this from defines to something else?

Reply:
From what I understand the feature you are describing is not a standard preprocessor feature, correct?

Reply:
I assumed since bevy has this kind of [functionality](https://bevyengine.org/news/bevy-0-6/#shader-preprocessor) and the call it preprocessing it would be standard functionality.

Reply:
Ah I understand.

Ok so I just opened a [new milestone for version 2.0](https://github.com/elyshaffir/wgsl_preprocessor/milestone/1) in which the functionality of conditional compilation with a list of defines (like bevy's) should be supported.
I suggest you open an issue there about it and for now we'll make it only work with `//!define` statements.

The reason for that is that I want to make the conditional compilation feature be more efficient and remove WGSL code that is not compiled in the initial parsing phase (currently done in `load_shader_module`), which would probably require breaking the existing functionality (hence the major version change).

Reply:
I'll also open a couple of issues there that I had in mind.

Reply:
What I suggest is modifying `load_shader_module` to ignore some lines similar to how you did it in `parse_defines` depending on which `//!define` statements and `//!ifdef` statements were already parsed.

Reply:
I added a new issue (I don't have access to add it to the milestones) with the proposed changes.
[#4](https://github.com/elyshaffir/wgsl_preprocessor/issues/4#issue-1473206762)

Reply:
Great, now be sure to integrate your changes into `load_shader_module` and we can merge.